src/components/select/Select.js

```diff
@@ -198,7 +198,7 @@ export default class SelectComponent extends ListComponent {
   }
 
   selectValueAndLabel(data) {
-    const value = this.getOptionValue((this.isEntireObjectDisplay() && !this.itemValue(data)) ? data : this.itemValue(data));
+    const value = this.getOptionValue((this.isEntireObjectDisplay() || !this.itemValue(data)) ? data : this.itemValue(data));
```

Comment:
I am afraid that can effect the select with EntireObject property value. Please discuss it with Hanna.

Reply:
Thanks for the comment I found another solution for my issue